firebase-ai-friendly-meals/apple/FriendlyMeals/FriendlyMeals/Features/MealPlannerChat/MealPlannerChatViewModel.swift

```diff
@@ -0,0 +1,240 @@
+//
+// FriendlyMeals
+//
+// Copyright © 2025 Google LLC.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import ConversationKit
+import FirebaseAI
+import SwiftUI
+
+@Observable
+@MainActor
+class MealPlannerChatViewModel {
+  var messages: [DefaultMessage] = []
+
+  private var timer: Timer?
+  private var remainingTime: Int = 0
+  private var isTimerRunning: Bool = false
```

Comment:
Could a consumer of this class observe the `timer` property and use `timer != nil` to know if the timer is running instead (should this be a computed property instead of manually storing and setting a bool on timer creation/invalidation)?